docs/guides/validation-user-guide.md

```diff
@@ -4,4 +4,461 @@ headline: Validation User Guide
+expression to match the field values. Also, some common pattern modifiers are available:
+ - `dot_all` (a.k.a. "single line") — enables the dot (`.`) symbol to match all the characters,
+   including line breaks;
+ - `case_insensitive` — allows to ignore the case of the matched symbols;
+ - `multiline` — enables the `^` (caret) and `$` (dollar) signs to match a start and an end of
+   a line instead of a start and an end of the whole expression;
+ - `unicode` — enables matching the whole UTF-8 sequences;
+ - `partial_match` — allows the matched strings to contain a full match to the pattern and some 
+   other characters as well. By default, a string only matches a pattern if it is a full match,
+   i.e. there are no unaccounted for leading and/or trailing characters.
+   
+Example:
+
+```proto
+import "spine/options.proto";
+
+// A link to an HTTP(S) resource.
+message HyperReference {
+    string url = 1 [(pattern) = { 
+            regex: "https?://.+\\..+" 
+            modifier: {
+                case_insensitive: true
+            }
+    }];
+}
+```
+
+It is recommended to use simple patterns due to performance considerations. For example, fully
+fledged URL and email patterns are famously too long to be used in most cases. Treat `(pattern)`
+checks as if they were yet another code with regex matching in it.
+
+### Temporal constraint
+
+Spine provides an option for validating time-bearing types. Those are:
+ - `google.protobuf.Timestamp`;
+ - `spine.time.YearMonth`;
+ - `spine.time.LocalDate`;
+ - `spine.time.LocalDateTime`;
+ - `spine.time.OffsetDateTime`;
+ - `spine.time.ZonedDateTime`;
+ - any user-defined type which implements the Temporal interface (`io.spine.time.Temporal` for
+   Java).
+   
+Using the option `(when)`, you may declare that the timestamp should lie in past or in future.
+
+```proto
+import "spine/time_options.proto";
+import "spine/time/time.proto";
+
+// A command to place an order.
+message PlaceOrder {
+    
+    // ...
+
+    spine.time.ZonedDateTime when_placed = 12 [(when).in = PAST];
+    spine.time.ZonedDateTime when_expires = 13 [(when).in = FUTURE];
+}
+```
+
+Note that the value is checked in relation to the current server time. In most cases, this should
+not be an issue. However, be aware that using `FUTURE` in Events and entity states may cause
+validation errors upon playing historical Events in future. However, that is not the case with
+Commands.  
+
+### Distinct collections
+
+Often, a `repeated` field logically represents a set rather than a list. Protobuf does not have
+a native support for sets. Moreover, it is often an invalid operation to add a duplicate element to
+a set. For such cases, Spine provides the `(distinct)` option, which constrains a `repeated` or
+a `map` field to only contain non-duplicating elements (values in case of `map`s).
+
+Example:
+
+```proto
+import "spine/options.proto";
+
+// The state of the User Aggregate.
+message User {
+
```

Comment:
This sample is not a complete definition of the `User` type. So, I think we should have `// ...` in the middle.